### PR TITLE
Fix organogram filenames

### PR DESCRIPTION
**Pull request by kentsanggds**
## What

It seems like changes to the ckan database and solr index are not picked up by Publish automatically, so in order to force it to pick up the filename fixes in this card - https://trello.com/c/5KrmZ1tb/1025-fix-bug-to-show-visualisation-on-junior-post-organograms, we need to run a one-off script to update the filenames in the Publish DB, then we just reindex elasticsearch and the filenames should be updated.

The input to the process was generated by a run of this script on the appropriate environment - 

https://github.com/alphagov/ckanext-datagovuk/blob/master/bin/python_scripts/fix_organograms_s3_filenames.py

### CODE REVIEW

File: lib/tasks/fix_existing_organogram_filenames.rake [hunk 1]
```diff
@@ -0,0 +1,6 @@
+require_relative 'update_organogram_filenames.rb'
+
+desc "Fix existing organogram filenames"
+task :fix_existing_organogram_filenames => :environment do
+  UpdateOrganogramFilenames.new().call
```
Comment (thomasleese) [inline lib/tasks/fix_existing_organogram_filenames.rake]: Generally in Ruby if a function doesn't take any arguments you would call it without the `()`.

```suggestion
  UpdateOrganogramFilenames.new.call
```

File: lib/tasks/update_organogram_filenames.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    parse_csv(csv_file)
+    msg = replace_urls()
+    puts msg
+
+  end
+
+  def parse_csv(csv_file)
+    puts "Parsing CSV '#{csv_file}'"
+    File.open(csv_file).readlines.each do |line|
```
Comment (thomasleese) [inline lib/tasks/update_organogram_filenames.rb]: You can read this using the CSV module.

```rb
CSV.foreach(csv_file) do |row|
  @old_urls << row.first
  @new_urls << row.last
end
```

File: lib/tasks/update_organogram_filenames.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    end
+  end
+
+  def replace_urls
+    puts "Searching for urls containing '-posts-'..."
+
+    if @old_urls.empty?
+      "No urls to process"
+    else
+      Link.all.each do |link|
```
Comment (thomasleese) [inline lib/tasks/update_organogram_filenames.rb]: If you're interested, using [`Link.find_each` is slightly more efficient here](https://api.rubyonrails.org/classes/ActiveRecord/Batches.html#method-i-find_each), but it probably don't make a big difference here.

File: lib/tasks/update_organogram_filenames.rb (hunk 2 above)
Comment (kentsanggds) [reply lib/tasks/update_organogram_filenames.rb]: I'll try that

**Pull request merged**